src/main/kotlin/net/djvk/fireflyPlaidConnector2/transactions/TransferMatcher.kt

```diff
@@ -0,0 +1,144 @@
+package net.djvk.fireflyPlaidConnector2.transactions
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import java.time.ZoneId
+import kotlin.math.abs
+
+/**
+ * Identifies transaction pairs that represent a transfer from one account into another.
+ */
+@Component
+class TransferMatcher(
+    @Value("\${fireflyPlaidConnector2.timeZone}")
+    private val timeZoneString: String,
+
+    @Value("\${fireflyPlaidConnector2.transferMatchWindowDays}")
+    private val transferMatchWindowDays: Long,
+) {
+    private val zoneId = ZoneId.of(timeZoneString)
+    private val transferMatchWindowSeconds = transferMatchWindowDays * 24 * 60 * 60
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    val transferTypes =
+        setOf(
+            PersonalFinanceCategoryEnum.Primary.TRANSFER_IN,
+            PersonalFinanceCategoryEnum.Primary.TRANSFER_OUT,
+            PersonalFinanceCategoryEnum.Primary.LOAN_PAYMENTS,
+            PersonalFinanceCategoryEnum.Primary.BANK_FEES,
+        )
+
+    data class SortByPairsResult(
+        val singles: List<PlaidFireflyTransaction>,
+        val pairs: List<Pair<PlaidFireflyTransaction, PlaidFireflyTransaction>>,
```

Comment:
Ah, I think this goes back [here](https://github.com/dvankley/firefly-plaid-connector-2/pull/96/files#r1659431460) where you called-out the overlap between the usage of the word `match`.

A `MatchedTransaction`, which contains both Plaid and Firefly data, is still only a single transaction. The purpose of the TransferMatcher is to find where we have two "transactions" (one deposit, one withdrawal) that can be combined into a "transfer". At the moment there is no `PlaidFireflyTransaction` subtype that can represent a transfer. I think this may be a gap in how I was thinking about transactions and I'll consider how to close it...

Reply:
Ok, I think I'm starting to better understand this.
A `MatchedTransaction` is not a pair of transactions that are candidates for combining into a transfer or are about to be combined into a transfer, it is a pair of objects representing the Firefly and Plaid versions of the same transaction and is intended to solve [this edge case](https://github.com/dvankley/firefly-plaid-connector-2/pull/96#discussion_r1660330776).

Reply:
And you say this pretty clearly [here](https://github.com/nprzy/firefly-plaid-connector-2/blob/9713978338a6b6631f3e38df5e60e4f92578ae34/src/main/kotlin/net/djvk/fireflyPlaidConnector2/transactions/PlaidFireflyTransaction.kt#L9), I just didn't understand it. I think I'm on board now.